src/pkg/cli/compose/loader.go

```diff
@@ -24,29 +25,23 @@ func (c Loader) LoadCompose(ctx context.Context) (*compose.Project, error) {
 	if err != nil {
 		return nil, err
 	}
-	term.Debug("Loading compose file", composeFilePath)
-	workDir := filepath.Dir(composeFilePath)
+	term.Debugf("Loading compose file %q", composeFilePath)
 
 	// Compose-go uses the logrus logger, so we need to configure it to be more like our own logger
 	logrus.SetFormatter(&logrus.TextFormatter{DisableTimestamp: true, DisableColors: !term.StderrCanColor(), DisableLevelTruncation: true})
 
-	projOpts, err := getDefaultProjectOptions(workDir)
+	projOpts, err := getDefaultProjectOptions(composeFilePath)
 	if err != nil {
 		return nil, err
 	}
 
-	// HACK: We do not want to include all the os environment variables, only COMPOSE_PROJECT_NAME
-	if envProjName, ok := os.LookupEnv("COMPOSE_PROJECT_NAME"); ok {
-		projOpts.Environment["COMPOSE_PROJECT_NAME"] = envProjName
-	}
-
```

Comment:
Though I cannot think of any actual behaviour difference, but personally I prefer keeping this hack here instead of using `cli.WithName` on line 97, just to keep the behaviour consistent.